src/SignalR/clients/ts/FunctionalTests/EchoConnectionHandler.cs

```diff
@@ -4,13 +4,21 @@
 using System.Buffers;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Http.Connections;
 
 namespace FunctionalTests
 {
     public class EchoConnectionHandler : ConnectionHandler
     {
         public async override Task OnConnectedAsync(ConnectionContext connection)
         {
+            var context = connection.GetHttpContext();
+            // The 'withCredentials' tests wont send a cookie for cross-site requests
+            if (!context.WebSockets.IsWebSocketRequest && !context.Request.Cookies.ContainsKey("testCookie"))
```

Comment:
Why do you need to check IsWebSocketRequest here?

Reply:
We have a test that checks if WebSockets works, and it wont have any cookies on it, plus we don't test WebSockets with `withCredentials = false` because that isn't a thing.

But, this might be the wrong check for WebSockets?

Reply:
This should be fine for checking WebSockets. The main quirk with this property is that if the WebSocket middleware **has not run yet** then this property will return `false` **even if the request is a WebSocket upgrade**. We talked about putting a default feature in that just did the detection (and threw on `Accept`), but I don't believe we ever did that.